object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -123,129 +119,89 @@ void checkMustCallInvoked(ControlFlowGraph cfg) {
-      // If the node is not skipped by the shouldSkipInvokePseudoAssignCheck() it means we have a
-      // method invocation or object creation node that creates a new resource, so we increment
-      // the numMustCall
-      incrementNumMustCall();
-      checkPseudoAssignToOwning(node);
+  private void handleInvocation(Set<ImmutableSet<LocalVarWithTree>> defs, Node node) {
+    doOwnershipTransferToParameters(defs, node);
+
+    if (shouldSkipInvokePseudoAssignCheck(node)) {
+      return;
     }
+    updateDefsWithTempVar(defs, node);
   }
 
-  /**
-   * Given a node representing a method or constructor call, checks that if the call has a non-empty
-   * {@code @MustCall} type, then its result is pseudo-assigned to some location that can take
-   * ownership of the result
-   */
-  private void checkPseudoAssignToOwning(Node node) {
+  private void updateDefsWithTempVar(Set<ImmutableSet<LocalVarWithTree>> defs, Node node) {
     Tree tree = node.getTree();
-    List<String> mustCallVal = typeFactory.getMustCallValue(tree);
-    if (mustCallVal.isEmpty()) {
-      return;
-    }
-    // Default to assuming that non-source nodes will always be assigned to
-    // an owning location, but that source nodes will not. This allows the checker
-    // to correctly handle foreach loops.
-    boolean assignedToOwning = !node.getInSource();
-    AssignmentContext assignmentContext = node.getAssignmentContext();
-    if (assignmentContext != null) {
-      Element elementForType = assignmentContext.getElementForType();
-      if (assignmentContext instanceof AssignmentLhsContext) {
-        // lhs should be a local variable
-        assignedToOwning = isOwningAssignmentLhs(elementForType);
-      } else if (assignmentContext instanceof MethodParameterContext) {
-        // must be an @Owning or @MustCallChoice parameter
-        assignedToOwning =
-            typeFactory.getDeclAnnotation(elementForType, Owning.class) != null
-                || typeFactory.hasMustCallChoice(elementForType);
-      } else if (assignmentContext instanceof MethodReturnContext) {
-        // must be an @Owning return
-        assignedToOwning =
-            TRANSFER_OWNERSHIP_AT_RETURN
-                ? typeFactory.getDeclAnnotation(
-                        TreeUtils.elementFromTree(assignmentContext.getContextTree()),
-                        NotOwning.class)
-                    == null
-                : typeFactory.getDeclAnnotation(elementForType, Owning.class) != null;
-      } else if (assignmentContext instanceof LambdaReturnContext) {
-        // TODO handle this case.  For now we will report an error
-      } else {
-        throw new BugInCF("unexpected AssignmentContext type " + assignmentContext.getClass());
+    if (typeFactory.mapTempVarToNode.inverse().containsKey(node.getTree())) {
+
+      LocalVariableNode temporaryLocal = typeFactory.mapTempVarToNode.inverse().get(node.getTree());
```

Comment:
Move this line before the `if` check, and change the `if` check to see if `temporaryLocal == null`